internal/service/connect/hours_of_operation.go

```diff
@@ -0,0 +1,336 @@
+package connect
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/connect"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+)
+
+func ResourceHoursOfOperation() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceHoursOfOperationCreate,
+		ReadContext:   resourceHoursOfOperationRead,
+		UpdateContext: resourceHoursOfOperationUpdate,
+		DeleteContext: resourceHoursOfOperationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(connectHoursOfOperationCreatedTimeout),
+			Delete: schema.DefaultTimeout(connectHoursOfOperationDeletedTimeout),
+		},
+		Schema: map[string]*schema.Schema{
+			"config": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"day": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice(connect.HoursOfOperationDays_Values(), false),
+						},
+						"end_time": {
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Required: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"hours": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+									"minutes": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+								},
+							},
+						},
+						"start_time": {
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Required: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"hours": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+									"minutes": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"description": {
```

Comment:
To add validation, min length 1 max length 250.

ValidateFunc: validation.StringLenBetween(1, 250),

Reply:
Resolved in af1e8156408209cc246b110dd3dac257787e239e